web/src/navbar_alerts.ts

```diff
@@ -173,6 +184,31 @@ export function toggle_organization_profile_incomplete_banner(): void {
     }
 }
 
+export async function should_show_upload_quota_banner(ls: LocalStorage): Promise<boolean> {
+    if (page_params.is_spectator) {
+        return false;
+    }
+
+    const space_used = await get_used_upload_space_percentage();
+    used_upload_space_percentage = Number(space_used);
+    const threshold = current_user.is_admin ? 90 : 95;
+    if (used_upload_space_percentage < threshold) {
+        return false;
+    }
+
+    if (!localstorage.supported() || ls.get("lastUpgradeQuotaLimitReminderTime") === undefined) {
+        return true;
+    }
+
+    const last_upload_quota_banner_dismissal_time = ls.get("lastUpgradeQuotaLimitReminderTime");
+    assert(typeof last_upload_quota_banner_dismissal_time === "number");
+    const upload_quota_banner_snooze_duration = addDays(
+        new Date(last_upload_quota_banner_dismissal_time),
+        7,
+    ).getTime();
+    return Date.now() > upload_quota_banner_snooze_duration;
```

Comment:
The `upload_quota_banner_snooze_duration` variable name is bad; it's not a duration, but instead the time after that duration. Maybe `upload_quota_banner_min_date` would be fine.